libs/shared/utils/src/lib/provenance.ts

```diff
@@ -1,17 +1,67 @@
 import { exec } from 'node:child_process';
 import { promisify } from 'node:util';
+import { withTimeout } from './utils';
 
-export async function nxLatestHasProvenance(): Promise<boolean> {
-  const npmView = (
-    await promisify(exec)(
-      `npm view nx@latest dist.attestations.provenance --json`,
-      {
+/**
+ * Checks the provenance of the latest version of Nx from the npm registry.
+ * Returns true if the provenance is valid, otherwise returns a string describing the failure reason.
+ */
+export async function nxLatestProvenanceCheck(): Promise<true | string> {
+  try {
+    const npmView = (
+      await promisify(exec)(`npm view nx@latest --json`, {
```

Comment:
This has a chance to log NPM warnings I think, which makes the output not valid JSON. Using `--silent` should work.